src/java.rs

```diff
@@ -124,16 +109,16 @@ pub fn deobfuscate_bytecode_signature(
 }
 
 /// formats types (param_type list, return_type) into a human-readable signature
-pub fn format_signature(types: &Option<(Vec<String>, String)>) -> Option<String> {
-    if types.is_none() {
-        return None;
-    }
+pub fn format_signature(types: Option<DeobfuscatedSignature>) -> Option<String> {
+    let types = types?;
 
-    let (parameter_java_types, return_java_type) = types.as_ref().unwrap();
+    let parameter_java_types = types.parameters_types();
+    let return_java_type = types.return_type();
 
-    let mut signature = format!("({})", parameter_java_types.join(", "));
+    let mut signature = format!("({})", parameter_java_types.collect::<Vec<_>>().join(", "));
```

Comment:
as this is all crate-internal, you do not need to use the iterator / collect roundtrip here.

The type and iterator interface was just to hide these implementation details for *outside* users of the crate.